packages/nx/src/hasher/task-hasher.ts

```diff
@@ -332,7 +332,10 @@ class TaskHasherImpl {
                     visited
                   );
                 } else {
-                  const hash = this.hashExternalDependency(d.target);
+                  const hash = this.hashExternalDependency(
+                    task.target.target,
```

Comment:
This should be source from the dependency. Task target would not be accurate, as it's a name of the target not a package.
```suggestion
                    d.source,
```